kedro/framework/cli/cli.py

```diff
@@ -138,13 +138,21 @@ def main(
             project_metadata=self._metadata, command_args=args
         )
 
-        super().main(
-            args=args,
-            prog_name=prog_name,
-            complete_var=complete_var,
-            standalone_mode=standalone_mode,
-            **extra,
-        )
+        try:
+            super().main(
+                args=args,
+                prog_name=prog_name,
+                complete_var=complete_var,
+                standalone_mode=standalone_mode,
+                **extra,
+            )
+        except SystemExit as exc:
+            self._cli_hook_manager.hook.after_command_run(  # pylint: disable=no-member
+                project_metadata=self._metadata, command_args=args, exit_code=exc.code
+            )
+            # click.core.main() method exits by default, we capture this and then
+            # exit as originally intended
```

Comment:
I'd suggest moving the comment up just before the `except`.

And just to clarify: `click.core.main()` will always throw a `SystemExit` and so we'll always execute the `except` block?

Reply:
Yeah basically every route of this exception handler rises `sys.exit({{code}})`